akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -349,12 +440,79 @@ class CircuitBreaker(
   private def resetBreaker(): Unit = transition(HalfOpen, Closed)
 
   /**
+   * Invokes all onSuccess callback handlers.
+   *
+   * @param start time in nanoseconds of call invocation
+   */
+  private def notifyCallSuccessListeners(start: Long): Unit = if (!successListeners.isEmpty) {
+    val elapsed = System.nanoTime() - start
+    val iterator = successListeners.iterator()
+    while (iterator.hasNext) {
+      val listener = iterator.next()
+      executor.execute(new Runnable {
+        def run() = listener.accept(elapsed)
+      })
```

Comment:
That's fine. I assume that the number of listeners won't be that great, so the savings of allocating on `Runnable` for all callbacks v.s. one for each callback aren't that big.